### PR TITLE
feat: update ascii , char_length and strpos functions definitions.

### PR DESCRIPTION
**Pull request by lviiii**
Thank you for submitting a PR!

Before you continue, please ensure that your commit messages follow [conventional commit syntax](1). This will save you a rebase later. Substrait uses an automated release process that, among other things, uses the commit messages to build a changelog, so the syntax and format matters!

The title of the PR should also be a valid commit header.

Some examples of proper commit message headers and PR titles:

 - `feat: add feature X`
 - `fix: X in case of Y`
 - `docs: improve documentation for X`

Note the case and grammar conventions.

Furthermore, any commit that imposes a breaking change should end in a paragraph that starts with `BREAKING CHANGE: ...`, where `...` explains what changed. The automated release process uses this to determine how it should bump the version number. Anything that changes the behavior of a plan that was previously legal is considered a breaking change; note that this includes behavior specifications that only exist in Substrait in the form of behavior descriptions on the website or in comments.

Please also note breaking changes in the first comment of the PR, so we're less likely to miss them when we review or merge.

[1]: https://www.conventionalcommits.org/en/v1.0.0/

### CODE REVIEW

File: extensions/functions_string.yaml [hunk 1]
```diff
@@ -17,6 +42,83 @@ scalar_functions:
...
+          - value: "string"
+            name: "substr"
+        return: i32
+      - args:
+          - value: "fixedchar<l1>"
+            name: "input"
+          - value: "fixedchar<l1>"
+            name: "substr"
+        return: i32
+  - name: length
```
Comment (richtia) [inline extensions/functions_string.yaml:104]: https://github.com/substrait-io/substrait/blob/main/extensions/functions_string.yaml#L943

File: extensions/functions_string.yaml [hunk 2]
```diff
@@ -17,6 +42,83 @@ scalar_functions:
         variadic:
           min: 1
         return: "string"
+  -
+    name: instr
```
[the remark this answers is not in the document]
Comment (lviiii) [reply extensions/functions_string.yaml]: Thx, I updated strpos and char_length functions definitions to add i32 type.

File: extensions/functions_string.yaml [hunk 3]
```diff
@@ -473,6 +498,138 @@ scalar_functions:
...
+      - args:
+          - name: case_sensitivity
+            options: [ CASE_SENSITIVE, CASE_INSENSITIVE, CASE_INSENSITIVE_ASCII ]
+            required: false
+          - value: "string"
+            name: "input"
+            description: The input string.
+          - value: "fixedchar<L2>"
+            name: "substring"
+            description: The substring to search for.
```
Comment (richtia) [inline extensions/functions_string.yaml:522]: The original MR kept the types consistent. Is there a particular reason you want to introduce the interaction between different types?

Here's the original comment about this: https://github.com/substrait-io/substrait/issues/251#issuecomment-1191249269

Comment (lviiii) [reply extensions/functions_string.yaml]: Thanks for your comments, and I kept the types consistent.

File: extensions/functions_string.yaml [hunk 4]
```diff
@@ -1,6 +1,31 @@
 %YAML 1.2
 ---
 scalar_functions:
+  -
+    name: ascii
+    description: Returns the numeric value of the first character of `input`.
```
Comment (richtia) [inline extensions/functions_string.yaml:6]:
```suggestion
    description: Returns the ASCII value of the first character of `input`.
```

Comment (richtia) [inline extensions/functions_string.yaml:6]: Could you also give more details about the UTF8 option?  The first 128 characters match up with ascii, but what happens after that when the input is UTF8?

**Comment by jacques-n** [comment]
No progress has been made in more than six months. Closing without prejudice.

**Pull request closed without merging**